cou/cli.py

```diff
@@ -251,4 +279,6 @@ def entrypoint() -> None:
         logger.exception(exc)
         sys.exit(2)
     finally:
+        if args.command == "upgrade":
+            loop.run_until_complete(_run_finalizer_command(args))
```

Comment:
For this, we could probably include the finaliser steps in the plan, and run the finaliser steps here only.  I think it would be confusing if it's part of the normal plan _and_ here.

Reply:
I think it will be nicer to have a separate output for pre-upgrade sanity check and post-upgrade sanity check that are outside the upgrade plan

Reply:
e.g.

```
Connected to 'ceph' ✔
Analyzing cloud... ✔
Running pre-upgrade sanity check...
Upgrade cloud from 'yoga' to 'zed'
    <some plan>
Running post-upgrade sanity check...
```

Reply:
> Users won't know of this support if we don't show it in the plan because it's not explicit.

User don't need to know about this (same for pre upgrade sanity check). I think we don't need to show it in the upgrade plan as well. The (pre or post) upgrade sanity check should be always run, and the users don't need to know what steps were ran (you can ofc display the steps for transparency because it doesn't hurt). It only matters if it's failed sanity check or not, and if yes, how do I fix it to make it happy. 

I think this is more of a UI and UX issue, than an actual code issue. Moreover, we have some issues with the current approach of running steps and showing the step (I will dm you about the ticket). Can you open an issue regarding this?

Reply:
If it's just a UI/UX issue, could we create a separate issue and continue the discussion there? Thanks!